File: uhrwerk-core/src/main/java/io/qimia/uhrwerk/config/representation/File.java
package io.qimia.uhrwerk.config.representation;

import io.qimia.uhrwerk.config.ConfigException;

public class File{
    private String path;

    public File(){}

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public void validate(String path){
        if(this.path==null){
            throw new ConfigException("Missing field: " + path + "path");
        }
    }
}

File: uhrwerk-config/src/main/kotlin/io/qimia/uhrwerk/config/representation/File.kt
package io.qimia.uhrwerk.config.representation

import io.qimia.uhrwerk.config.builders.ConfigException

data class File(
    override var name: String? =null,
    var path: String? = null
) : Connection() {
    override fun validate(path: String) {
        //super.validate(path)
        if (this.path == null) {
            throw ConfigException("Missing field: " + path + "path")
        }
    }
}